### PR TITLE
Fix to set error for initial LE creation if errors are shown

### PR DESCRIPTION
**Pull request by iliajie**
This PR will set an error to be accessible and shown outside of `create_initial_letsencrypt_cert` sub.

### CODE REVIEW

File: virtual-server-lib-funcs.pl [hunk 1]
```diff
@@ -8556,8 +8556,10 @@ sub create_initial_letsencrypt_cert
 				push(@estr, $e->{'desc'}." : ".
 					    &html_strip($e->{'error'}));
 				}
-			&$second_print(&text('letsencrypt_evalid',
-				&html_escape(join(", ", @estr))));
+			my $err = &html_escape(join(", ", @estr));
+			$virtual_server::create_initial_letsencrypt_cert::error =
```
Comment (jcameron) [inline virtual-server-lib-funcs.pl:8560]: Don't use global variables like this to pass the error around - it's bad programming practice. Instead just have it returned from the `create_initial_letsencrypt_cert` function.

Comment (iliajie) [reply virtual-server-lib-funcs.pl:8560]: I know. But It has to be returned by create domain function, which would involve more significant changes.

Comment (jcameron) [reply virtual-server-lib-funcs.pl:8560]: Best to put it into a field in the domain hash then. Using global variables as a "secret" return value is never a good idea..

**Comment by iliajie** [comment]
> Best to put it into a field in the domain hash then.

Agreed. Check this https://github.com/virtualmin/virtualmin-gpl/pull/847 PR instead!

**Pull request closed without merging**